### PR TITLE
Add integration tests for JMX.

### PR DESCRIPTION
**Pull request by musa-asad**
### Description of changes
Adds integration tests to check if enabling annotations adds expected environmental variables.

### Testing
`make test`
Screenshot 2024-10-09 at 9 40 03 AM

See passing tests in `Checks`: https://github.com/aws/amazon-cloudwatch-agent-operator/actions/runs/11259284939?pr=244.

----------------
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/operator-integration-test.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ on:
     types: [ opened, reopened, synchronize, ready_for_review ]
     branches:
       - main
+      - jmx-merge
```
Comment (musa-asad) [inline .github/workflows/operator-integration-test.yml:13]: Note: REMOVE.

File: integration-tests/jmx/default_instrumentation_jmx_env_variables.json
Comment (Paramadon) [inline integration-tests/jmx/default_instrumentation_jmx_env_variables.json:1]: Are these the only defaul instrumentation of jmx?

File: integration-tests/jmx/sample-deployment-jmx.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+spec:
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+      annotations:
+        instrumentation.opentelemetry.io/inject-java: "true"
```
Comment (Paramadon) [inline integration-tests/jmx/sample-deployment-jmx.yaml:14]: Why are you injecting java for jmx deployment?

File: integration-tests/jmx/default_instrumentation_jmx_env_variables.json
Comment (musa-asad) [reply integration-tests/jmx/default_instrumentation_jmx_env_variables.json:1]: [Yes](https://github.com/aws/amazon-cloudwatch-agent-operator/pull/240/files#diff-a85df2e3cfbd396ebc189457e323b6b1811fd93e540b564d8e9180cef6813b3aR162-R164).

**Review by Paramadon — APPROVED** [review]

**Review by JayPolanco — APPROVED** [review]

**Pull request merged**